app/models/host/base.rb

```diff
@@ -172,6 +172,11 @@ def set_interfaces(parser)
         iface = get_interface_scope(name, attributes).first || interface_class(name).new(:managed => false)
         # create or update existing interface
         set_interface(attributes, name, iface)
+        if iface.primary?
+          self.interfaces[0] = iface
```

Comment:
I can find it in the array if that's possible, but the moment I clear the array it does not work.